### PR TITLE
Do not list templates without a file in the document from template form

### LINKED ISSUE #2360: Fehler bei "Dokument aus Vorlage" wenn Dokument nur in Papierform vorhanden ist
Es gibt ein Fehler bei "Dokument aus Vorlage" wenn Dokument nur in Papierform vorhanden ist

screen3

### PR DESCRIPTION
**Pull request by tarnap**
do not list templates without digitally available document

Closes  #2360

### CODE REVIEW

File: docs/HISTORY.txt [hunk 1]
```diff
@@ -5,6 +5,7 @@ Changelog
 2017.3.0 (unreleased)
 ---------------------
 
+- Do not display templates if its document is not digitally available. [tarnap]
```
Comment (phgross) [inline docs/HISTORY.txt]: It's not clear where the empty templates are not listed ... 

Suggestion:
` - Do not list empty templates (without a file) in the CreateDocumentFromTemplate form.`

File: opengever/dossier/tests/test_templatefolder.py [hunk 2]
```diff
@@ -484,6 +486,26 @@ def test_doc_properties_are_not_created_when_disabled(self, browser):
         with TemporaryDocFile(document.file) as tmpfile:
             self.assertItemsEqual([], read_properties(tmpfile.path))
 
+    @browsing
+    def test_templates_without_digitally_available_document_not_listed(self, browser):
```
Comment (phgross) [inline opengever/dossier/tests/test_templatefolder.py]: `test_templates_without_a_file_are_not_listed`

File: opengever/dossier/tests/test_templatefolder.py [hunk 3]
```diff
@@ -484,6 +486,26 @@ def test_doc_properties_are_not_created_when_disabled(self, browser):
...
+        create(Builder('document')
+               .titled('Template with no content')
```
Comment (phgross) [inline opengever/dossier/tests/test_templatefolder.py]: Title needs to be unicode:

`.titled(u'Template with no content')`

File: opengever/dossier/tests/test_templatefolder.py [hunk 4]
```diff
@@ -484,6 +486,26 @@ def test_doc_properties_are_not_created_when_disabled(self, browser):
...
+               .within(self.templatefolder)
+               .with_modification_date(datetime(2010, 12, 28)))
```
Comment (phgross) [inline opengever/dossier/tests/test_templatefolder.py]: not necessary ...

File: opengever/dossier/tests/test_templatefolder.py [hunk 5]
```diff
@@ -484,6 +486,26 @@ def test_doc_properties_are_not_created_when_disabled(self, browser):
...
+
+        browser.login().open(self.dossier, view='document_with_template')
+        self.assertEquals(
```
Comment (phgross) [inline opengever/dossier/tests/test_templatefolder.py]: I would simplify the assertion here

```
self.assertEquals(
    ['Template A', 'Template B'], 
    [row.get('Title') for row in browser.css('table.listing').first.dicts()])
```

File: opengever/dossier/tests/test_templatefolder.py (hunk 3 above)
Comment (tarnap) [inline opengever/dossier/tests/test_templatefolder.py]: All other implementations in the same test file use standard strings.
Shall I create am issue to correct these as well?
https://github.com/4teamwork/opengever.core/blob/master/opengever/dossier/tests/test_templatefolder.py#L103
https://github.com/4teamwork/opengever.core/blob/master/opengever/dossier/tests/test_templatefolder.py#L146
https://github.com/4teamwork/opengever.core/blob/master/opengever/dossier/tests/test_templatefolder.py#L234
https://github.com/4teamwork/opengever.core/blob/master/opengever/dossier/tests/test_templatefolder.py#L306
https://github.com/4teamwork/opengever.core/blob/master/opengever/dossier/tests/test_templatefolder.py#L380
https://github.com/4teamwork/opengever.core/blob/master/opengever/dossier/tests/test_templatefolder.py#L433
https://github.com/4teamwork/opengever.core/blob/master/opengever/dossier/tests/test_templatefolder.py#L453
https://github.com/4teamwork/opengever.core/blob/master/opengever/dossier/tests/test_templatefolder.py#L474

**Comment by phgross** [comment]
@tarnap bitte rebasen ...

**Comment by tarnap** [comment]
@phgross check

**Pull request merged**